FirebaseStorageSwift/Sources/StorageError.swift

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseStorage
+
+public enum StorageError: Error {
+  case unknown
+  case objectNotFound
+  case bucketNotFound
+  case projectNotFound
+  case quotaExceeded
+  case unauthenticated
+  case unauthorized
+  case retryLimitExceeded
+  case nonMatchingChecksum
+  case downloadSizeExceeded
+  case cancelled
+  case invalidArgument
+  case internalError
```

Comment:
Are there any of these that contain extra metadata in the `userInfo` dictionary? I could see potentially the `invalidArgument` or something having the argument passed back, or the project / bucket  info, download size max, etc.

Reply:
Yep. Good call. I'm working through an update.